examples/stream-hello-world/StreamHelloWorld_Client.cpp

```diff
@@ -26,20 +28,22 @@ int main(int argc, char* argv[]) {
   {
     // this example runs inside the Future.then lambda
     LOG(INFO) << "------------------ Run in future.then";
-    auto s = std::make_shared<ExampleSubscriber>(5, 6);
+    auto s = yarpl::Reference<yarpl::Subscriber<Payload>>(
+        new ExampleSubscriber(5, 6));
     rsf->connect().then([s](std::shared_ptr<RSocketRequester> rs) {
-      rs->requestStream(Payload("Bob"), s);
+      rs->requestStream(Payload("Bob"))->subscribe(s);
     });
-    s->awaitTerminalEvent();
+    //    s->awaitTerminalEvent();
```

Comment:
I can't make `Reference<ExampleSubscriber>` and still pass in to the `subscribe(s)` call. 

What do I need to do so I can still have it be `ExampleSubscriber` and call methods such as `awaitTerminalEvent`?

My expectation is that `Reference` allows refCounting similar to `shared_ptr`. Am I misunderstanding? Does it actually behave like `unique_ptr` so I need to move it into `subscribe`?